qiskit/transpiler/passes/calibration/rzx_builder.py

```diff
@@ -164,15 +164,21 @@ def get_calibration(self, node_op: CircuitInst, qubits: List) -> Union[Schedule,
             schedule: The calibration schedule for the RZXGate(theta).
 
         Raises:
+            QiskitError: if rotation angle is not assigned.
             QiskitError: If the control and target qubits cannot be identified.
             CalibrationNotAvailable: RZX schedule cannot be built for input node.
         """
         theta = node_op.params[0]
 
+        try:
+            theta = float(theta)
+        except TypeError as ex:
+            raise QiskitError("Target rotation angle is not assigned.") from ex
+
         rzx_theta = Schedule(name="rzx(%.3f)" % theta)
         rzx_theta.metadata["publisher"] = CalibrationPublisher.QISKIT
 
-        if np.isclose(theta, 0.0):
+        if np.isclose(float(theta), 0.0):
```

Comment:
Good point! I think the explicit cast is better than the one in `np.isclose()` (which answers your question in the affirmative), and also note these are treated differently between the `RZXCalibrationBuilder` and `RZXCalibrationBuilderNoEcho`. I think it might be also worth an effort to combine these classes for ease of future maintenance.

Reply:
Sure, I agree we can do further refactoring down the road. For a bugfix release this is a good way to scope this. Let me just remove this redudant cast here and we can merge this for 0.22.1 and save the larger refactoring for 0.23.0:

 ```suggestion
         if np.isclose(theta, 0.0):
```